### ISSUE TITLE
PCores and SylowSubgroups are not known after computing SylowSystem for nilpotent groups

### ISSUE THREAD

**Issue by hungaborhorvath** [gap-system/gap#412]
I do not want to overdo this whole "set attribute thing," so just let me know if I should stop. 

I found that if SylowSystem is computed for a nilpotent group, then the Sylow Subgroups are not known to GAP, neither are PCores. Not even for the ones occurring in the SylowSystem. Then calling PCores can be costly, depending on the group. 

```
A := DihedralGroup(16);; 
B := SmallGroup(27, 3);; 
C := SmallGroup(125, 4);; 
D := DirectProduct(A, B, C, SmallGroup(1536, 2));;
F := DirectProduct(D, D, D, D);;
G := DirectProduct(F, F, F, F);; 
gap> HasIsNilpotentGroup(G); 
true
gap> IsNilpotentGroup(G);
true
gap> SylowSystem(G); time; 
[ <pc group with 208 generators>, <pc group with 64 generators>, 
  <pc group with 48 generators> ]
544
gap> ComputedSylowSubgroups(G);
[  ]
gap> ComputedPCores(G);
[  ]
gap> PCore(G, 2); time; 
<pc group of size 
411376139330301510538742295639337626245683966408394965837152256 with 
208 generators> 
18352
```

Calling SylowSubgroups is apparently not costly, because it uses the same pcgs method as the SylowSystem, thus all of the things needed are computed already. In fact, it took 0 time to compute any of the Sylow Subgroups, but only after that did GAP update the ComputedSylowSubgroups list. However, the two groups have different known attributes, since the one in SylowSystem has no size, whereas the one computed with SylowSubgroup does have. 

Maybe a new PCore method should be written for at least nilpotent groups and/or if some Sylow subgroups are computed by SylowSystem? Currently it seems that Core (called from grppc.gi:1029) is really slow. `IsNormal(G, SylowSubgroup(G,2));` would run in 1272 rather than in 18352, thus it might be worth to force an IsNormal check. 

IMO, though, ComputedPCores and ComputedSylowSubgroups should be updated after SylowSystem (or only SylowSubgroup) is computed for nilpotent groups. However, I do not know the proper syntax to update an appropriate KeyDependentOperation without actually calling it. And it might be tricky to figure out which Sylow subgroup in SylowSystem correspond to which time without having computed the sizes of these Sylow subgroups, which is an argument for updating these attributes while computing SylowSystem for nilpotent groups rather than just writing a new method.

**Comment by fingolfin** [comment]
Has this been resolved by #535?

**Comment by hungaborhorvath** [comment]
@fingolfin Yes, it has, I just forgot to close it at the time. I will do it now.
